### ISSUE TITLE
Raspberry Pi Compute Module error  mmc0: command never completed

### ISSUE THREAD

**Issue by JiapengLi** [raspberrypi/linux#1317]
I am using the Raspbian Jessie 2016-02-26 image, and I got some mmc errors during the system booting. When return back to wheezy 2014-12-2, the issue disappears.

Update 2016-03-03:
To reproduce the issue, extra usb Ethernet module should be connected to Compute Module IO Board. I tried two USB Ethernet adapter(one smsc95xx and another XiaoMi), both have the issue.

```
Raspbian GNU/Linux 8 raspberrypi ttyAMA0

raspberrypi login: Uncompressing Linux... done, booting the kernel.
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Initializing cgroup subsys cpuset
[    0.000000] Initializing cgroup subsys cpu
[    0.000000] Initializing cgroup subsys cpuacct
[    0.000000] Linux version 4.1.18+ (dc4@dc4-XPS13-9333) (gcc version 4.9.3 (crosstool-NG crosstool-ng-1.22.0-88-g8460611) ) #846 Thu Feb 25 14:11:56 GMT 2016
[    0.000000] CPU: ARMv6-compatible processor [410fb767] revision 7 (ARMv7), cr=00c5387d
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT nonaliasing instruction cache
[    0.000000] Machine model: Raspberry Pi Compute Module Rev 1.0
[    0.000000] cma: Reserved 8 MiB at 0x1b400000
[    0.000000] Memory policy: Data cache writeback
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 113680
[    0.000000] Kernel command line: dma.dmachans=0x7f35 bcm2708_fb.fbwidth=656 bcm2708_fb.fbheight=416 bcm2708.boardrev=0x14 bcm2708.serial=0xd694b121 smsc95xx.macaddr=B8:27:EB:94:B1:21 bcm2708_fb.fbswap=1 bcm2708.uart_clock=3000000 vc_mem.mem_base=0x1ec00000 vc_mem.mem_size=0x20000000  dwc_otg.lpm_enable=0 console=ttyAMA0,115200 console=tty1 root=/dev/mmcblk0p2 rootfstype=ext4 elevator=deadline fsck.repair=yes rootwait
[    0.000000] PID hash table entries: 2048 (order: 1, 8192 bytes)
[    0.000000] Dentry cache hash table entries: 65536 (order: 6, 262144 bytes)
[    0.000000] Inode-cache hash table entries: 32768 (order: 5, 131072 bytes)
[    0.000000] Memory: 436836K/458752K available (5792K kernel code, 488K rwdata, 1788K rodata, 352K init, 720K bss, 13724K reserved, 8192K cma-reserved)
[    0.000000] Virtual kernel memory layout:
[    0.000000]     vector  : 0xffff0000 - 0xffff1000   (   4 kB)
[    0.000000]     fixmap  : 0xffc00000 - 0xfff00000   (3072 kB)
[    0.000000]     vmalloc : 0xdc800000 - 0xff000000   ( 552 MB)
[    0.000000]     lowmem  : 0xc0000000 - 0xdc000000   ( 448 MB)
[    0.000000]     modules : 0xbf000000 - 0xc0000000   (  16 MB)
[    0.000000]       .text : 0xc0008000 - 0xc076f1f0   (7581 kB)
[    0.000000]       .init : 0xc0770000 - 0xc07c8000   ( 352 kB)
[    0.000000]       .data : 0xc07c8000 - 0xc0842260   ( 489 kB)
[    0.000000]        .bss : 0xc0842260 - 0xc08f63e0   ( 721 kB)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS:522
[    0.000000] clocksource stc: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 1911260446275 ns
[    0.000015] sched_clock: 32 bits at 1000kHz, resolution 1000ns, wraps every 2147483647500ns
[    0.000056] Switching to timer-based delay loop, resolution 1000ns
[    0.000345] Console: colour dummy device 80x30
[    0.001271] console [tty1] enabled
[    0.001328] Calibrating delay loop (skipped), value calculated using timer frequency.. 2.00 BogoMIPS (lpj=10000)
[    0.001412] pid_max: default: 32768 minimum: 301
[    0.001790] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.001860] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.002911] Initializing cgroup subsys blkio
[    0.003005] Initializing cgroup subsys memory
[    0.003075] Initializing cgroup subsys devices
[    0.003137] Initializing cgroup subsys freezer
[    0.003199] Initializing cgroup subsys net_cls
[    0.003318] CPU: Testing write buffer coherency: ok
[    0.003437] ftrace: allocating 19747 entries in 58 pages
[    0.107910] Setting up static identity map for 0x81c0 - 0x81f8
[    0.110088] devtmpfs: initialized
[    0.120934] VFP support v0.3: implementor 41 architecture 1 part 20 variant b rev 5
[    0.121388] clocksource jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.123380] pinctrl core: initialized pinctrl subsystem
[    0.124279] NET: Registered protocol family 16
[    0.129773] DMA: preallocated 4096 KiB pool for atomic coherent allocations
[    0.131274] bcm2708.uart_clock = 3000000
[    0.136147] hw-breakpoint: found 6 breakpoint and 1 watchpoint registers.
[    0.136224] hw-breakpoint: maximum watchpoint size is 4 bytes.
[    0.136432] Serial: AMBA PL011 UART driver
[    0.136734] 20201000.uart: ttyAMA0 at MMIO 0x20201000 (irq = 83, base_baud = 0) is a PL011 rev2
[    0.505638] console [ttyAMA0] enabled
[    0.510035] bcm2835-mbox 2000b880.mailbox: mailbox enabled
[    0.556094] bcm2708-dmaengine 20007000.dma: DMA legacy API manager at f2007000, dmachans=0xf35
[    0.564823] bcm2708-dmaengine 20007000.dma: Initialized 7 DMA channels (+ 1 legacy)
[    0.573393] bcm2708-dmaengine 20007000.dma: Load BCM2835 DMA engine driver
[    0.580322] bcm2708-dmaengine 20007000.dma: dma_debug:0
[    0.586477] SCSI subsystem initialized
[    0.590638] usbcore: registered new interface driver usbfs
[    0.596357] usbcore: registered new interface driver hub
[    0.601885] usbcore: registered new device driver usb
[    0.607910] raspberrypi-firmware soc:firmware: Attached to firmware from 2016-02-25 14:25
[    0.643599] Switched to clocksource stc
[    0.696803] FS-Cache: Loaded
[    0.700105] CacheFiles: Loaded
[    0.718844] NET: Registered protocol family 2
[    0.724789] TCP established hash table entries: 4096 (order: 2, 16384 bytes)
[    0.731982] TCP bind hash table entries: 4096 (order: 2, 16384 bytes)
[    0.738614] TCP: Hash tables configured (established 4096 bind 4096)
[    0.745251] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.751149] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.757843] NET: Registered protocol family 1
[    0.762816] RPC: Registered named UNIX socket transport module.
[    0.768873] RPC: Registered udp transport module.
[    0.773648] RPC: Registered tcp transport module.
[    0.778380] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    0.786095] hw perfevents: enabled with armv6_1176 PMU driver, 3 counters available
[    0.795234] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.817699] VFS: Disk quotas dquot_6.6.0
[    0.822083] VFS: Dquot-cache hash table entries: 1024 (order 0, 4096 bytes)
[    0.831768] FS-Cache: Netfs 'nfs' registered for caching
[    0.838540] NFS: Registering the id_resolver key type
[    0.843835] Key type id_resolver registered
[    0.848060] Key type id_legacy registered
[    0.855929] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 252)
[    0.863919] io scheduler noop registered
[    0.867919] io scheduler deadline registered (default)
[    0.873470] io scheduler cfq registered
[    0.879905] BCM2708FB: allocated DMA memory 5b800000
[    0.885079] BCM2708FB: allocated DMA channel 0 @ f2007000
[    0.895634] Console: switching to colour frame buffer device 82x26
[    0.906887] Serial: 8250/16550 driver, 0 ports, IRQ sharing disabled
[    0.916646] vc-cma: Videocore CMA driver
[    0.922338] vc-cma: vc_cma_base      = 0x00000000
[    0.928787] vc-cma: vc_cma_size      = 0x00000000 (0 MiB)
[    0.935863] vc-cma: vc_cma_initial   = 0x00000000 (0 MiB)
[    0.943160] vc-mem: phys_addr:0x00000000 mem_base=0x1ec00000 mem_size:0x20000000(512 MiB)
[    0.973767] brd: module loaded
[    0.987594] loop: module loaded
[    0.993390] vchiq: vchiq_init_state: slot_zero = 0xdb880000, is_master = 0
[    1.003969] Loading iSCSI transport class v2.0-870.
[    1.012078] usbcore: registered new interface driver smsc95xx
[    1.019748] dwc_otg: version 3.00a 10-AUG-2012 (platform bus)ethernet
[    1.227425] Core Release: 2.80a
[    1.232115] Setting default values for core params
[    1.238492] Finished setting default values for core params
[    1.445903] Using Buffer DMA mode
[    1.450805] Periodic Transfer Interrupt Enhancement - disabled
[    1.458257] Multiprocessor Interrupt Enhancement - disabled
[    1.465473] OTG VER PARAM: 0, OTG VER FLAG: 0
[    1.471459] Dedicated Tx FIFOs mode
[    1.477051] WARN::dwc_otg_hcd_init:1047: FIQ DMA bounce buffers: virt = 0xdb814000 dma = 0x5b814000 len=9024
[    1.490211] FIQ FSM acceleration enabled for :
[    1.490211] Non-periodic Split Transactions
[    1.490211] Periodic Split Transactions
[    1.490211] High-Speed Isochronous Endpoints
[    1.513547] WARN::hcd_init_fiq:412: FIQ on core 0 at 0xc03dd6ac
[    1.521129] WARN::hcd_init_fiq:413: FIQ ASM at 0xc03dd988 length 36
[    1.529052] WARN::hcd_init_fiq:438: MPHI regs_base at 0xdc896000
[    1.536782] dwc_otg 20980000.usb: DWC OTG Controller
[    1.543407] dwc_otg 20980000.usb: new USB bus registered, assigned bus number 1
[    1.552447] dwc_otg 20980000.usb: irq 32, io mem 0x00000000
[    1.559695] Init: Port Power? op_state=1
[    1.565223] Init: Power Port (0)
[    1.570323] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002
[    1.578783] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    1.587641] usb usb1: Product: DWC OTG Controller
[    1.593953] usb usb1: Manufacturer: Linux 4.1.18+ dwc_otg_hcd
[    1.601262] usb usb1: SerialNumber: 20980000.usb
[    1.608487] hub 1-0:1.0: USB hub found
[    1.613982] hub 1-0:1.0: 1 port detected
[    1.620651] usbcore: registered new interface driver usb-storage
[    1.628759] mousedev: PS/2 mouse device common for all mice
[    1.636987] bcm2835-cpufreq: min=700000 max=700000
[    1.643793] sdhci: Secure Digital Host Controller Interface driver
[    1.651545] sdhci: Copyright(c) Pierre Ossman
[    1.657895] sdhost: log_buf @ db813000 (5b813000)
[    1.743678] mmc0: sdhost-bcm2835 loaded - DMA enabled (>1)
[    1.751325] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.779558] ledtrig-cpu: registered to indicate activity on CPUs
[    1.787527] hidraw: raw HID events driver (C) Jiri Kosina
[    1.794912] usbcore: registered new interface driver usbhid
[    1.802082] usbhid: USB HID core driver
[    1.812030] Initializing XFRM netlink socket
[    1.818200] Indeed it is in host mode hprt0 = 00021501
[    1.885504] NET: Registered protocol family 17
[    1.891956] Key type dns_resolver registered
[    1.904657] mmc0: command never completed.
[    1.910376] mmc0:>cmd op 1 arg 0x0 flags 0xe1 - resp 00000000 00000000 00000000 00000000, err 0
[    1.922169] mmc0: =========== REGISTER DUMP ===========
[    1.929036] mmc0: SDCMD  0x00004001
[    1.934118] mmc0: SDARG  0x00000000
[    1.939150] mmc0: SDTOUT 0x00030d40
[    1.944128] mmc0: SDCDIV 0x0000026f
[    1.949061] mmc0: SDRSP0 0x40ff8080
[    1.953949] mmc0: SDRSP1 0x00003f7f
[    1.958812] mmc0: SDRSP2 0xffffffff
[    1.963617] mmc0: SDRSP3 0xffffffff
[    1.968377] mmc0: SDHSTS 0x00000010
[    1.973091] mmc0: SDVDD  0x00000001
[    1.977746] mmc0: SDEDM  0x00010800
[    1.982358] mmc0: SDHCFG 0x0000040a
[    1.986931] mmc0: SDHBCT 0x00000000
[    1.991471] mmc0: SDHBLC 0x00000000
[    1.995960] mmc0: ===========================================
[    2.003367] registered taskstats version 1
[    2.009105] vc-sm: Videocore shared memory driver
[    2.015088] [vc_sm_connected_init]: start
[    2.021273] [vc_sm_connected_init]: end - returning 0
[    2.029880] Waiting for root device /dev/mmcblk0p2...
[    2.124022] mmc0: command never completed.
[    2.129156] mmc0:>cmd op 1 arg 0x40200000 flags 0xe1 - resp 00000000 00000000 00000000 00000000, err 0
[    2.140533] mmc0: =========== REGISTER DUMP ===========
[    2.146898] mmc0: SDCMD  0x00004001
[    2.151535] mmc0: SDARG  0x40200000
[    2.156171] mmc0: SDTOUT 0x00024978
[    2.160810] mmc0: SDCDIV 0x00000340
[    2.165461] mmc0: SDRSP0 0xc0ff8080
[    2.170112] mmc0: SDRSP1 0x00003f7f
[    2.174769] mmc0: SDRSP2 0xffffffff
[    2.179417] mmc0: SDRSP3 0x3fe0203f
[    2.184070] mmc0: SDHSTS 0x00000010
[    2.188730] mmc0: SDVDD  0x00000001
[    2.193395] mmc0: SDEDM  0x00010800
[    2.198022] mmc0: SDHCFG 0x0000040a
[    2.202597] mmc0: SDHBCT 0x00000000
[    2.207137] mmc0: SDHBLC 0x00000000
[    2.211632] mmc0: ===========================================
[    2.218741] usb 1-1: new high-speed USB device number 2 using dwc_otg
[    2.226450] mmc0: error -5 whilst initialising MMC card
[    2.234069] Indeed it is in host mode hprt0 = 00001101
[    2.355168] mmc0: command never completed.
[    2.360393] mmc0:>cmd op 1 arg 0x0 flags 0xe1 - resp 00000000 00000000 00000000 00000000, err 0
[    2.371211] mmc0: =========== REGISTER DUMP ===========
[    2.377597] mmc0: SDCMD  0x00004001
[    2.382242] mmc0: SDARG  0x00000000
[    2.386884] mmc0: SDTOUT 0x000186a0
[    2.391520] mmc0: SDCDIV 0x000004e0
[    2.396149] mmc0: SDRSP0 0x40ff8080
[    2.400791] mmc0: SDRSP1 0x00003f7f
[    2.405434] mmc0: SDRSP2 0xffffffff
[    2.410072] mmc0: SDRSP3 0xffffffff
[    2.414701] mmc0: SDHSTS 0x00000010
[    2.419333] mmc0: SDVDD  0x00000001
[    2.423963] mmc0: SDEDM  0x00010800
[    2.428590] mmc0: SDHCFG 0x0000040a
[    2.433238] mmc0: SDHBCT 0x00000000
[    2.437839] mmc0: SDHBLC 0x00000000
[    2.442375] mmc0: ===========================================
[    2.464096] usb 1-1: New USB device found, idVendor=0424, idProduct=9512
[    2.472014] usb 1-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    2.481564] hub 1-1:1.0: USB hub found
[    2.486704] hub 1-1:1.0: 3 ports detected
[    2.546372] mmc0: command never completed.
[    2.551513] mmc0:>cmd op 1 arg 0x0 flags 0xe1 - resp 00000000 00000000 00000000 00000000, err 0
[    2.562233] mmc0: =========== REGISTER DUMP ===========
[    2.568589] mmc0: SDCMD  0x00004001
[    2.573212] mmc0: SDARG  0x00000000
[    2.577825] mmc0: SDTOUT 0x0000ee88
[    2.582436] mmc0: SDCDIV 0x000007fd
[    2.587055] mmc0: SDRSP0 0x40ff8080
[    2.591682] mmc0: SDRSP1 0x00003f7f
[    2.596303] mmc0: SDRSP2 0xffffffff
[    2.600929] mmc0: SDRSP3 0xffffffff
[    2.605550] mmc0: SDHSTS 0x00000010
[    2.610173] mmc0: SDVDD  0x00000001
[    2.614791] mmc0: SDEDM  0x00010800
[    2.619422] mmc0: SDHCFG 0x0000040a
[    2.624011] mmc0: SDHBCT 0x00000000
[    2.628537] mmc0: SDHBLC 0x00000000
[    2.633019] mmc0: ===========================================
[    2.813668] usb 1-1.1: new high-speed USB device number 3 using dwc_otg
[    2.934061] usb 1-1.1: New USB device found, idVendor=0424, idProduct=ec00
[    2.942144] usb 1-1.1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    2.953755] smsc95xx v1.0.4
[    3.018134] smsc95xx 1-1.1:1.0 eth0: register 'smsc95xx' at usb-20980000.usb-1.1, smsc95xx USB 2.0 Ethernet, b8:27:eb:94:b1:21
[   70.584955] random: nonblocking pool is initialized

```

**Comment by pelwell** [comment]
Because sdhost is still a relatively new driver which has had a major revision recently, there is more logging output under error conditions than might otherwise be the case.

My understanding is that an MMC device will return an error to CMD1 (SEND_OP_COND) until it is ready, and that this can take 100s of milliseconds.It is also possible that it is slow to respond and that the timeout in the new driver is too short. However, polling until the command succeeds is valid behaviour.

If these few initial CMD1 timeouts ("command never completed", with SDCMD set to 0x00004001) are the only errors you are seeing then I don't think you need to be concerned. I'll either make sure the messages disappear in a future update.

**Comment by JiapengLi** [comment]
@pelwell Thank you for your reply. My issue is when I get these errors Compute Module stops starting, 
the firmware stuck at some place. The log I pasted is the full log I could get during the error starting, after that it stops.

**Comment by dpslwk** [comment]
I've also seen the same on Compute Modules with the jessie-lite 26th image
Plug in a USB to ethernet adapter and boot stalls with logs very similar to above, 
also seen it on occasion with a USB Wifi dongle (RT5370)

**Comment by pelwell** [comment]
I recommend you add `dtoverlay=mmc` to your config.txt, which will revert to using the old driver and interface, until the problem is better understood.

**Comment by bggardner** [comment]
I've had the same error message(s) with multiple Compute Modules.  Some CMs would boot, then error after a number of reboots, and others would error at first boot.  It appears to be worse in the 2016-03-18 image, but not confirmed.  Adding `dtoverlay=mmc` to `/boot/config.txt` as @pelwell suggested seems to have fixed the issue (thanks!).

**Comment by ghost** [comment]
I have the same problem with external I2C device connected to the raspberry pi compute module with jessie-lite(2016-03-18). Adding dtoverlay=mmc to /boot/config.txt  has fixed the issue.

**Comment by benpye** [comment]
I appear to be hitting this same issue when trying to use the sdio overlay and reverting the eMMC interface to mmc. Has there been any progress on this issue? I am currently being blocked by this as I require an SD card for user accessible files.

https://www.raspberrypi.org/forums/viewtopic.php?f=98&t=151655

**Comment by pelwell** [comment]
Can you explain why you think you can't use the sdhost driver on the external SD interface? Yes SDIO using sdhost is problematic because it aggressively clock gates the SD clock, and for that reason SDIO support has never been tested on it which means it's almost certainly broken, but a regular SD card should be just fine. If you can't figure out the DT overlay needed, create a new issue including what you've done so far and I'll take a look. Alternatively you could go straight to a Pull Request, with the understanding that there will be some back-and-forth before the finished overlay becomes part of the standard distribution.

Driving the eMMC through SDHOST appears to be failing because an early timeout is marginal, and I haven't yet found a way to increase it. I'm hoping to get a bit of Broadcom support on this.

**Comment by Ruffio** [comment]
@JiapengLi has your issue been resolved? If so, please close this issue. Thanks.

**Comment by JiapengLi** [comment]
@Ruffio Thank you for notification. It is solved.

**Comment by mildred** [comment]
I'm getting this issue with ArchLinux ARM (don't know when it was last updated, this iw a read only filesystem. It appears at bootup and blocks boot. I have a 4G/LTE device connected when this happens. If I disconnect the usb port (the LTE/4G dongle) the boot completes correctly and adding the dongle later in boot corrects the problem.

**Comment by JamesH65** [comment]
Does this happen with Raspbian? We don't support Arch, but if it happens on Raspbian then we can take a look. Also need to know the type and vendor ID of the USB device.

**Comment by mildred** [comment]
the usb 4G dongle is a Huawei E3372 (12d1:14dc)

**Comment by JamesH65** [comment]
Does it still go wrong under Raspbian?

**Comment by JamesH65** [comment]
If you hot plug the device (ie plug in after boot), what does lsusb and dmsg say?

**Comment by mildred** [comment]
Sorry, I haven't tried. I changed the SD card for a new one and I don't have the problem any more. Perhaps the SD card was too old. The filesystem on it was mounted read-only to avoid damaging the SD card too early, but it didn't work well enough. Previously, I had a number of problems with linux system on flash devices when not mounted read only. The flash device (sd card and usb drive, both) would corrupt really quickly.
